### ISSUE TITLE
labels disappears using render_highquality

### ISSUE THREAD

**Issue by blueyeti1994** [tylermorganwall/rayshader#313]
The labels works fine when i use plot_3d and the name appears in map. But when i want use render_highquality, only box rectangular black box appears with no text. How to solve this

`
render_label(elev_matrix, lat =27.663,
             long =85.29765, extent=attr(elev_img, "extent"),
             freetype=T, zscale = 6, text = "My label",
             textcolor = "Black"
             dashed = F, textsize = 1, linewidth = 2,
             linecolor="Darkorange")

outfile <- "image/final_plt.png"

  render_highquality(
    filename = outfile,
    interactive = FALSE,
    lightdirection = 120,
    lightaltitude = c(20, 80),
    lightcolor = c("white"),
    lightintensity = c(600, 100),
    line_radius = 1, 
    samples = 150,
    width = 1200,
    height = 600
  )
`
please find both working image and image i'm having issue whenusing render_highquality

![traces](https://github.com/tylermorganwall/rayshader/assets/170682504/f3df45ba-a44b-4e05-8719-466f49553432)
![reference](https://github.com/tylermorganwall/rayshader/assets/170682504/89901daf-7987-4cab-bc57-5128de4ec48e)

**Comment by vgXhc** [comment]
I can confirm this. I'm on MacOS (Sonoma 14.6.1 (23G93))
Here's the (non-reproducible) example that led me to discover this.

![image](https://github.com/user-attachments/assets/c1af4e6d-0d5a-44ab-b04e-84236219ca09)

```
  plot_3d(heightmap = bw_el_matrix,
          zscale = 25,
          windowsize = c(1920,1080),
          water = TRUE,
          watercolor="imhof2")

#function that creates multiple labels from the cities data frame
create_labels <- function(city_id){
  render_label(heightmap = bw_el_matrix,
               lat = st_coordinates(cities[city_id,])[2],
               long = st_coordinates(cities[city_id,])[1],
               extent = boundary_projected,
               textcolor = "white", 
               dashed = TRUE,
               linecolor = "lightgrey",
               text = cities_day_1_sf$city[city_id],
               zscale = 25)
}
walk(1:nrow(cities), create_labels)
render_snapshot()

render_highquality(filename = "day_1.png",
                   text_size = 20,
                   samples = 512)
```

It also replicated when using the `render_label()` code [from the documentation](https://www.rayshader.com/reference/render_label.html).

```
> sessionInfo()
R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.6.1

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] tmap_3.3-4       osmdata_0.2.5    lubridate_1.9.3  forcats_1.0.0   
 [5] stringr_1.5.1    purrr_1.0.2      readr_2.1.5      tidyr_1.3.1     
 [9] tibble_3.2.1     tidyverse_2.0.0  spData_2.3.3     rgl_1.3.12      
[13] rayshader_0.38.1 ggplot2_3.5.1    dplyr_1.1.4      raster_3.6-30   
[17] sp_2.1-4         terra_1.7-78     sf_1.0-16       

loaded via a namespace (and not attached):
 [1] DBI_1.2.2               httr2_1.0.1             tmaptools_3.1-1        
 [4] rlang_1.1.4             magrittr_2.0.3          rayrender_0.35.0       
 [7] e1071_1.7-14            compiler_4.3.1          png_0.1-8              
[10] vctrs_0.6.5             pkgconfig_2.0.3         crayon_1.5.3           
[13] fastmap_1.2.0           magick_2.8.5            lwgeom_0.2-14          
[16] leafem_0.2.3            utf8_1.2.4              tzdb_0.4.0             
[19] xfun_0.49               satellite_1.0.5         jsonlite_1.8.9         
[22] progress_1.2.3          mapview_2.11.2          rayimage_0.11.0        
[25] parallel_4.3.1          prettyunits_1.2.0       R6_2.5.1               
[28] stringi_1.8.4           RColorBrewer_1.1-3      car_3.1-3              
[31] extrafontdb_1.0         jquerylib_0.1.4         stars_0.6-5            
[34] Rcpp_1.0.13-1           iterators_1.0.14        knitr_1.48             
[37] base64enc_0.1-3         extrafont_0.19          leaflet.providers_2.0.0
[40] timechange_0.3.0        tidyselect_1.2.1        rstudioapi_0.16.0      
[43] dichromat_2.0-0.1       abind_1.4-5             doParallel_1.0.17      
[46] codetools_0.2-19        curl_5.2.2              crsuggest_0.4          
[49] lattice_0.21-8          leafsync_0.1.0          withr_3.0.2            
[52] units_0.8-5             proxy_0.4-27            xml2_1.3.6             
[55] pillar_1.9.0            carData_3.0-5           KernSmooth_2.23-21     
[58] stats4_4.3.1            rayvertex_0.11.4        foreach_1.5.2          
[61] generics_0.1.3          hms_1.1.3               munsell_0.5.1          
[64] scales_1.3.0            class_7.3-22            glue_1.8.0             
[67] tools_4.3.1             XML_3.99-0.16.1         grid_4.3.1             
[70] Rttf2pt1_1.3.12         crosstalk_1.2.1         colorspace_2.1-1       
[73] tidygeocoder_1.0.5      Formula_1.2-5           cli_3.6.3              
[76] rappdirs_0.3.3          fansi_1.0.6             viridisLite_0.4.2      
[79] gtable_0.3.5            digest_0.6.37           classInt_0.4-10        
[82] htmlwidgets_1.6.4       farver_2.1.2            htmltools_0.5.8.1      
[85] lifecycle_1.0.4         leaflet_2.2.2           httr_1.4.7   
```

**Comment by vgXhc** [comment]
Investigating this further, I ran the example code on my Windows 10 machine. No issue with the `render_highquality()` there. So maybe it's something related to MacOS/XQuartz?

![test](https://github.com/user-attachments/assets/342f63d9-20a8-40e3-a1e0-f29cef1f9736)


```
> session_info()
─ Session info ─────────────────────────────────────
 setting  value
 version  R version 4.3.1 (2023-06-16 ucrt)
 os       Windows 10 x64 (build 19045)
 system   x86_64, mingw32
 ui       RStudio
 language (EN)
 collate  English_United States.utf8
 ctype    English_United States.utf8
 tz       America/Chicago
 date     2024-11-27
 rstudio  2023.06.0+421 Mountain Hydrangea (desktop)
 pandoc   3.1.1 @ C:/Program Files/RStudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)

─ Packages ─────────────────────────────────────────
 package                * version date (UTC) lib source
 base64enc                0.1-3   2015-07-28 [1] CRAN (R 4.3.0)
 cachem                   1.0.8   2023-05-01 [1] CRAN (R 4.3.1)
 cli                      3.6.2   2023-12-11 [1] CRAN (R 4.3.3)
 codetools                0.2-19  2023-02-01 [2] CRAN (R 4.3.1)
 colorspace               2.1-0   2023-01-23 [1] CRAN (R 4.3.1)
 crayon                   1.5.2   2022-09-29 [1] CRAN (R 4.3.1)
 curl                     5.2.1   2024-03-01 [1] CRAN (R 4.3.3)
 devtools               * 2.4.5   2022-10-11 [1] CRAN (R 4.3.1)
 digest                   0.6.33  2023-07-07 [1] CRAN (R 4.3.1)
 doParallel               1.0.17  2022-02-07 [1] CRAN (R 4.3.1)
 dplyr                    1.1.4   2023-11-17 [1] CRAN (R 4.3.3)
 ellipsis                 0.3.2   2021-04-29 [1] CRAN (R 4.3.1)
 evaluate                 0.24.0  2024-06-10 [1] CRAN (R 4.3.3)
 extrafont                0.19    2023-01-18 [1] CRAN (R 4.3.0)
 extrafontdb              1.0     2012-06-11 [1] CRAN (R 4.3.0)
 fansi                    1.0.6   2023-12-08 [1] CRAN (R 4.3.3)
 fastmap                  1.1.1   2023-02-24 [1] CRAN (R 4.3.1)
 foreach                  1.5.2   2022-02-02 [1] CRAN (R 4.3.1)
 fs                       1.6.2   2023-04-25 [1] CRAN (R 4.3.1)
 generics                 0.1.3   2022-07-05 [1] CRAN (R 4.3.1)
 glue                     1.6.2   2022-02-24 [1] CRAN (R 4.3.1)
 gsubfn                   0.7     2018-03-16 [1] CRAN (R 4.3.2)
 hms                      1.1.3   2023-03-21 [1] CRAN (R 4.3.1)
 htmltools                0.5.8.1 2024-04-04 [1] CRAN (R 4.3.3)
 htmlwidgets              1.6.4   2023-12-06 [1] CRAN (R 4.3.3)
 httpuv                   1.6.11  2023-05-11 [1] CRAN (R 4.3.1)
 httr                     1.4.7   2023-08-15 [1] CRAN (R 4.3.1)
 iterators                1.0.14  2022-02-05 [1] CRAN (R 4.3.1)
 janitor                  2.2.0   2023-02-02 [1] CRAN (R 4.3.1)
 jsonlite                 1.8.8   2023-12-04 [1] CRAN (R 4.3.3)
 knitr                    1.47    2024-05-29 [1] CRAN (R 4.3.3)
 later                    1.3.1   2023-05-02 [1] CRAN (R 4.3.1)
 lattice                  0.21-8  2023-04-05 [2] CRAN (R 4.3.1)
 lifecycle                1.0.4   2023-11-07 [1] CRAN (R 4.3.3)
 lubridate                1.9.3   2023-09-27 [1] CRAN (R 4.3.3)
 magick                   2.8.3   2024-02-18 [1] CRAN (R 4.3.3)
 magrittr                 2.0.3   2022-03-30 [1] CRAN (R 4.3.1)
 memoise                  2.0.1   2021-11-26 [1] CRAN (R 4.3.1)
 mime                     0.12    2021-09-28 [1] CRAN (R 4.3.0)
 miniUI                   0.1.1.1 2018-05-18 [1] CRAN (R 4.3.1)
 munsell                  0.5.1   2024-04-01 [1] CRAN (R 4.3.3)
 paws.common              0.7.3   2024-05-16 [1] CRAN (R 4.3.3)
 paws.security.identity * 0.6.1   2024-05-16 [1] CRAN (R 4.3.3)
 pillar                   1.9.0   2023-03-22 [1] CRAN (R 4.3.1)
 pkgbuild                 1.4.4   2024-03-17 [1] CRAN (R 4.3.3)
 pkgconfig                2.0.3   2019-09-22 [1] CRAN (R 4.3.1)
 pkgload                  1.3.4   2024-01-16 [1] CRAN (R 4.3.3)
 png                      0.1-8   2022-11-29 [1] CRAN (R 4.3.0)
 prettyunits              1.2.0   2023-09-24 [1] CRAN (R 4.3.1)
 priceR                   1.0.1   2023-10-22 [1] CRAN (R 4.3.2)
 profvis                  0.3.8   2023-05-02 [1] CRAN (R 4.3.1)
 progress                 1.2.3   2023-12-06 [1] CRAN (R 4.3.3)
 promises                 1.2.0.1 2021-02-11 [1] CRAN (R 4.3.1)
 proto                    1.0.0   2016-10-29 [1] CRAN (R 4.3.2)
 purrr                    1.0.2   2023-08-10 [1] CRAN (R 4.3.2)
 R6                       2.5.1   2021-08-19 [1] CRAN (R 4.3.1)
 raster                 * 3.6-26  2023-10-14 [1] CRAN (R 4.3.3)
 rayimage                 0.10.0  2023-12-09 [1] CRAN (R 4.3.3)
 rayrender                0.32.2  2024-02-20 [1] CRAN (R 4.3.3)
 rayshader              * 0.37.3  2024-02-21 [1] CRAN (R 4.3.3)
 rayvertex                0.10.4  2023-12-16 [1] CRAN (R 4.3.3)
 Rcpp                     1.0.11  2023-07-06 [1] CRAN (R 4.3.1)
 remotes                  2.5.0   2024-03-17 [1] CRAN (R 4.3.3)
 rgl                      1.3.1   2024-03-05 [1] CRAN (R 4.3.3)
 rlang                    1.1.4   2024-06-04 [1] CRAN (R 4.3.3)
 rmarkdown                2.27    2024-05-17 [1] CRAN (R 4.3.3)
 rstudioapi               0.16.0  2024-03-24 [1] CRAN (R 4.3.3)
 Rttf2pt1                 1.3.12  2023-01-22 [1] CRAN (R 4.3.0)
 scales                   1.3.0   2023-11-28 [1] CRAN (R 4.3.3)
 sessioninfo              1.2.2   2021-12-06 [1] CRAN (R 4.3.1)
 shiny                    1.8.1.1 2024-04-02 [1] CRAN (R 4.3.3)
 snakecase                0.11.1  2023-08-27 [1] CRAN (R 4.3.1)
 sp                     * 2.1-4   2024-04-30 [1] CRAN (R 4.3.3)
 stringi                  1.7.12  2023-01-11 [1] CRAN (R 4.3.0)
 stringr                  1.5.1   2023-11-14 [1] CRAN (R 4.3.3)
 terra                    1.7-78  2024-05-22 [1] CRAN (R 4.3.3)
 tibble                   3.2.1   2023-03-20 [1] CRAN (R 4.3.1)
 tidyselect               1.2.1   2024-03-11 [1] CRAN (R 4.3.3)
 timechange               0.3.0   2024-01-18 [1] CRAN (R 4.3.3)
 urlchecker               1.0.1   2021-11-30 [1] CRAN (R 4.3.1)
 usethis                * 2.2.3   2024-02-19 [1] CRAN (R 4.3.3)
 utf8                     1.2.4   2023-10-22 [1] CRAN (R 4.3.3)
 vctrs                    0.6.5   2023-12-01 [1] CRAN (R 4.3.3)
 xfun                     0.45    2024-06-16 [1] CRAN (R 4.3.3)
 xtable                   1.8-4   2019-04-21 [1] CRAN (R 4.3.1)
 yaml                     2.3.8   2023-12-11 [1] CRAN (R 4.3.2)

 [1] C:/Users/user1/AppData/Local/R/win-library/4.3
 [2] C:/Program Files/R/R-4.3.1/library

────────────────────────────────────────────────────
```

**Comment by faraksoch1881** [comment]
I'm on windows 11 and i have this issue when using redner_highquality.

**Comment by tylermorganwall** [comment]
Thank you for the bug report--the issue was  in the underlying rayimage package and has been fixed in 206b27fbd17809772d3c47d7f911b97aca1be464. Update rayimage to the latest version on github and it should be fixed.
